apps/content/api/public/recitations.py

```diff
@@ -41,15 +38,37 @@ class RecitationFilter(FilterSchema):
 @ordering(ordering_fields=["name", "created_at", "updated_at"])
 @searching(search_fields=["name", "description", "resource__publisher__name", "reciter__name"])
 def list_recitations(request, filters: RecitationFilter = Query()):
-    qs = Asset.objects.select_related("resource", "reciter").filter(
+    lang = request.LANGUAGE_CODE  # "ar" or "en"
+    qs = Asset.objects.select_related(
+        "resource", "resource__publisher", "reciter", "riwayah"
+    ).filter(
         category=Asset.CategoryChoice.RECITATION,
         resource__category=Resource.CategoryChoice.RECITATION,
         resource__status=Resource.StatusChoice.READY,
     )
 
     qs = filters.filter(qs)
-
-    # Ensure required slug comes from related Resource. Temp solution until we add slug to Asset model.
-    qs = qs.annotate(slug=F("resource__slug"))
+    qs = qs.annotate(
```

Comment:
we are using django-modelstranslation to avoid doing this hacks
just declare the `Publisher.name` field in translations.py 
and the package will handle the translations
also we should not be using `JSONObject` here, this should be delegated to a nested Schema

Reply:
unless there is something I am not understanding correctly